trunk/src/app/srs_app_dvr.cpp

```diff
@@ -642,7 +642,39 @@ srs_error_t SrsDvrPlan::on_video(SrsSharedPtrMessage* shared_video, SrsFormat* f
     if (!dvr_enabled) {
         return err;
     }
-    
+
+    // skip any SEI type nalus
+    if (format->vcodec->id == SrsVideoCodecIdAVC) {
+        for (int i = 0; i < format->video->nb_samples; ++i) {
+            SrsSample* sample = &format->video->samples[i];
+            SrsAvcNaluType avc_nalu_type;
+
+            if ((err = SrsVideoFrame::parse_avc_nalu_type(sample, avc_nalu_type)) != srs_success) {
+                return srs_error_wrap(err, "parse avc nalu_type");
+            }
+
+            if (avc_nalu_type == SrsAvcNaluTypeSEI) {
+                return err;
+            }
```

Comment:
If the video frame is [SEI P], will it drop? SEI always appears in Harward encoders like video toolbox, Nvidia enc, etc.

Reply:
fixed on #4201